homeassistant/components/sensor/mychevy.py

```diff
@@ -0,0 +1,155 @@
+"""Support for MyChevy sensors."""
+
+import asyncio
+from logging import getLogger
+from datetime import datetime as dt
+
+from homeassistant.components.mychevy import (
+    EVSensorConfig, DOMAIN, MYCHEVY_ERROR, MYCHEVY_SUCCESS,
+    NOTIFICATION_ID, NOTIFICATION_TITLE
+)
+from homeassistant.components.sensor import ENTITY_ID_FORMAT
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import (slugify)
+
+
+def battery_levels(entity):
+    """Compute battery levels for charge percentage."""
+    state = int(entity.state / 10)
+    if state == 10:
+        return "mdi:battery"
+    if state >= 1 and state < 10:
+        return "mdi:battery-%d0" % state
+    if state < 1:
+        return "mdi:battery-alert"
+
+
+SENSORS = [
+    EVSensorConfig("Mileage", "mileage", "miles", "mdi:speedometer"),
+    EVSensorConfig("Range", "range", "miles", "mdi:speedometer"),
+    EVSensorConfig("Charging", "charging"),
+    EVSensorConfig("Charge Mode", "charge_mode"),
+    EVSensorConfig("EVCharge", "percent", "%", "mdi:battery", battery_levels)
+]
+
+_LOGGER = getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the MyChevy sensors."""
+    if discovery_info is None:
+        return
+
+    hub = hass.data[DOMAIN]
+    sensors = [MyChevyStatus(hub)]
+    for sconfig in SENSORS:
+        sensors.append(EVSensor(hub, sconfig))
+
+    add_devices(sensors)
+
+
+class MyChevyStatus(Entity):
+    """A string representing the charge mode."""
+
+    _name = "MyChevy Status"
+    _icon = "mdi:car-connected"
+
+    def __init__(self, connection):
+        """Initialize sensor with car connection."""
+        self._state = None
+        self._last_update = dt.now()
+        self._conn = connection
```

Comment:
This isn't used. You could refactor this entity to fetch the connection status from `self._conn` instead of the connection instance calling either of the methods `success` or `error` on this entity. The former might be more natural and in line with the rest of the sensors here and also the rest of the sensors in home assistant.